test/over_react/component/error_boundary_test.dart

```diff
@@ -123,13 +133,30 @@ void main() {
         expect(jacket.getNode().text, 'hi there');
       });
 
+      test('its child when `state.error` is true and `state.showFallbackUIOnError` is false', () {
+        jacket = mount(ErrorBoundary()(dummyChild));
+        final component = jacket.getDartInstance();
+        component.setState(component.newState()
+          ..hasError = true
+          ..showFallbackUIOnError = false
+        );
+
+        expect(jacket.getNode().text, 'hi there');
+      });
+
       group('fallback UI when `state.error` is true', () {
-        test('', () {
+        test('and `state.showFallbackUIOnError` is true ("unrecoverable" error state)', () {
           jacket = mount(ErrorBoundary()(dummyChild));
+          expect(getByTestId(jacket.getInstance(), 'dummyChild'), isNotNull);
           final component = jacket.getDartInstance();
+          component.setState(component.newState()
+            ..hasError = true
+            ..showFallbackUIOnError = true
+          );
 
-          // Using throws for now since this is temporary, and the throwsUnimplementedError doesn't work here for some reason
-          expect(() => component.setState(component.newState()..hasError = true), throws);
+          expect(getByTestId(jacket.getInstance(), 'dummyChild'), isNull,
+              reason: 'The child component tree should have been removed from the dom');
+          expect(jacket.getNode(), hasAttr(defaultTestIdKey, 'ErrorBoundary.unrecoverableErrorInnerHtmlContainerNode'));
```

Comment:
Can we assert here that the static HTML got rendered? I'm not seeing that tested anywhere

Reply:
It is tested in the "gracefully handles errors in its tree when `props.fallbackUIRenderer` is not set" section: 

https://github.com/Workiva/over_react/pull/350/files#diff-820fada62e348ea7fccc19d5cc72d0f6R342

Reply:
Ah, sorry, I missed that. Thanks!